webide: remove trademark images and phrases

bolek-wisniewski opened the pull request:
criminal css hijacking of the monaco web editor, including, but probably
not limited to
* svg image replacement for logo
* welcome screen styling and content

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

bolek-wisniewski commented on web-ide/proxy/src/config.json:
```diff
@@ -2,7 +2,8 @@
         "port" : 3000,
-        "managementPort" : 3001
+        "managementPort" : 3001,
+        "httpToHttpsPort" : 3002
```
odd but the gcp loadbalancer being put in place does not have good redirect support, so I was told to open up another port for redirection. The load balancer will direct https traffic to our application by http:80 and direct http traffic to http:3002 so that the application can redirect it to https.

neil-da replied:
Good comment. Can we get it in the code?

bolek-wisniewski replied:
can't comment json - it's partially documented in httpToHttps.ts but I'll add to that

cocreature commented on web-ide/proxy/src/routes/httpToHttps.ts:
```diff
@@ -0,0 +1,26 @@
+        this.redirectApp.get('*', (req,res) => {
+            if(!req.secure) {
+                const r = ['https://', req.hostname, ":", 443, req.url].join('')
+                return res.redirect(307, r);
```
Why `307` rather than `301`, i.e., why a temporary redirect instead of a permanent redirect?

neil-da replied:
Because we might change our mind in future and don't want browsers to go straight to the new location. A permanent redirect is often a bad idea.

neil-da replied:
Unconvinced we won't go behind an Nginx style redirector, in which case we'd then ditch having https on this server at al.

bolek-wisniewski replied:
I noticed odd behavior in chrome without the temporary. It started redirecting things I didn't want, like other ports on the same host!

neil-da commented on web-ide/proxy/Dockerfile:
```diff
@@ -21,6 +21,6 @@ RUN apk update \
     && npm run compile
 
-EXPOSE 80 443 3001
+EXPOSE 80 443 3001 3002
```
Could we add comments as to what runs on each port? 80 and 443 I can guess. The others are less clear.

bolek-wisniewski replied:
👍

bolek-wisniewski commented on web-ide/proxy/static/images/logo.svg:
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
I grabbed it from daml.com and turned it into svg

neil-da replied:
From what? PNG? We can probably find a native SVG one somewhere.

bolek-wisniewski commented on web-ide/proxy/src/routes/webide.ts:
```diff
@@ -52,6 +55,12 @@ export default class WebIdeRoute {
 
+    handleIdeCss(req :Request, res :Response, next: NextFunction) {
+        const f = `${this.rootDir}/static/css/ide.main.css`
+        console.log("lets use our own css for %s : %s", req.url, f)
```
👍

bolek-wisniewski commented:
> Agree strongly that CSS file is too large to add to the repo - we won't have a chance of keeping up with it. Use CSS file importing and `!important` annotations to leave it as the original one plus a diff (that is applied at runtime by the browser).

@neil-da @moritzkiefer-da - I haven't spent too many cycles trying to improve this, but there are some technical annoyances. I'm not injecting myself into the browser lifecycle with my own javascript etc...that would be a better option but I haven't explored it yet...so I don't know how to load our own css other than intercepting the call to the css. I can

> Agree strongly that CSS file is too large to add to the repo - we won't have a chance of keeping up with it. Use CSS file importing and `!important` annotations to leave it as the original one plus a diff (that is applied at runtime by the browser).

thanks @neil-da @moritzkiefer-da - I've implemented a solution where we don't store all the code-server/vscode css. I intercept the call to the css and append our styles. Their css call has some weird hash in the url so I felt it would be better to serve up their content on the fly instead of static linking from our css.

let me know if the changes look OK. I've also removed some menu items and other buttons and added more text to the landing page

![Screen Shot 2019-04-26 at 6 43 19 PM](https://user-images.githubusercontent.com/25302921/56839986-34dd3f80-6853-11e9-8001-742f4cc153bc.png)

neil-da commented:
I still see 13K lines being added in the form of a huge CSS. I thought you said we weren't doing that anymore?

bolek-wisniewski commented:
> I still see 13K lines being added in the form of a huge CSS. I thought you said we weren't doing that anymore?

late friday brain fart...removed the unused css

neil-da commented on web-ide/proxy/static/css/webide.main.css:
```diff
@@ -0,0 +1,101 @@
+
+
+/*************************** Hide activity bar items **********************************************/
+.part.activitybar ul.actions-container li:nth-child(3),
```
What do these correspond to? Do they have aria-labels? That's quite a nice way of documenting it.

bolek-wisniewski replied:
unfortunately, those aria-labels have system specific / special characters in them. Chose this more concrete selector instead of the hassle and unknown (I'll add a comment)

neil-da commented on web-ide/proxy/static/css/webide.main.css:
```diff
@@ -0,0 +1,101 @@
+  \A\
+Troubleshooting \A\
+      The DAML language server log output is available under the \"Output\" panel (View->Output). \A\
+      Select \"DAML Language Server\" from the dropdown in the panel to see the log.";
```
Do we want the troubleshooting section for the user to see? My guess is it's more for us rather than users.

bolek-wisniewski replied:
will remove it 👍
